homeassistant/components/home_connect/light.py

```diff
@@ -0,0 +1,207 @@
+"""Provides a light for Home Connect."""
+import logging
+from math import ceil
+
+from homeconnect.api import HomeConnectError
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_HS_COLOR,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR,
+    LightEntity,
+)
+import homeassistant.util.color as color_util
+
+from .const import (
+    BSH_AMBIENT_LIGHT_BRIGHTNESS,
+    BSH_AMBIENT_LIGHT_COLOR,
+    BSH_AMBIENT_LIGHT_COLOR_CUSTOM_COLOR,
+    BSH_AMBIENT_LIGHT_CUSTOM_COLOR,
+    BSH_AMBIENT_LIGHT_ENABLED,
+    COOKING_LIGHTING,
+    COOKING_LIGHTING_BRIGHTNESS,
+    DOMAIN,
+)
+from .entity import HomeConnectEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Home Connect light."""
+
+    def get_entities():
+        """Get a list of entities."""
+        entities = []
+        hc_api = hass.data[DOMAIN][config_entry.entry_id]
+        for device_dict in hc_api.devices:
+            entity_dicts = device_dict.get("entities", {}).get("light", [])
+            entity_list = [HomeConnectLight(**d) for d in entity_dicts]
+            entities += entity_list
+        return entities
+
+    async_add_entities(await hass.async_add_executor_job(get_entities), True)
+
+
+class HomeConnectLight(HomeConnectEntity, LightEntity):
+    """Light for Home Connect."""
+
+    def __init__(self, device, desc, ambient):
+        """Initialize the entity."""
+        super().__init__(device, desc)
+        self._state = None
+        self._brightness = None
+        self._hs_color = None
+        self._ambient = ambient
+        if self._ambient:
+            self._brightnesskey = BSH_AMBIENT_LIGHT_BRIGHTNESS
+            self._key = BSH_AMBIENT_LIGHT_ENABLED
+            self._customcolorkey = BSH_AMBIENT_LIGHT_CUSTOM_COLOR
+            self._colorkey = BSH_AMBIENT_LIGHT_COLOR
+        else:
+            self._brightnesskey = COOKING_LIGHTING_BRIGHTNESS
+            self._key = COOKING_LIGHTING
+            self._customcolorkey = None
+            self._colorkey = None
+
+    @property
+    def is_on(self):
+        """Return true if the light is on."""
+        return bool(self._state)
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def hs_color(self):
+        """Return the color property."""
+        return self._hs_color
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        if self._ambient:
+            return SUPPORT_BRIGHTNESS | SUPPORT_COLOR
+        return SUPPORT_BRIGHTNESS
+
+    async def async_turn_on(self, **kwargs):
+        """Switch the light on, change brightness, change color."""
+        if self._ambient:
+            if ATTR_BRIGHTNESS in kwargs or ATTR_HS_COLOR in kwargs:
+                try:
+                    await self.hass.async_add_executor_job(
```

Comment:
Maybe this is a comment for another PR, but it seems wasteful to use our async api when the library interface needs to run in sync context.

We should probably adjust the entity parent class in a followup  PR so that we can update state with the sync api `self.schedule_update_ha_state`.

Reply:
OK